### PR TITLE
Stop clearing locked fields when the prefs are accessed by calling `settings.load()`

### LINKED ISSUE #5007: Deployment profile locked settings become unlocked after first launch
### Actual Behavior

When created locked deployment profile (either on Windows or Darwin), opening the application after launching and closing it, causes all the settings to be unlocked and changeable (dockerd Moby can be changed to containerd)

### Steps to Reproduce

Install Rancher Desktop, setup locked and default deployment profiles, launch Rancher desktop, close the window and reopen Rancher Desktop. When you reopen the application, the various settings that were locked before, are no longer locked.

### Result

After opening the Rancher application the second time, all of the settings are unlocked and able to be changed.

### Expected Behavior

The expected behavior is that the settings should still be locked down no matter how many times the application window is closed.

### Additional Information

_No response_

### Rancher Desktop Version

1.9.0

### Rancher Desktop K8s Version

1.27.1?

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

MacOS 13.4

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #5007 

This change actually implements the comment before the function.

### CODE REVIEW

**Comment by ericpromislow** [comment]
Marks' comment from the wrong PR #5015:

> I'm kind of worried that we're overloading load in a way that is confusing. Do we have a follow-up (not for the branch) to clean this up?

No we don't, although this change makes `settings.load` less destructive, as it should only modify globals and the file system the first time it's run now.

**Comment by jandubois** [comment]
> No we don't, although this change makes `settings.load` less destructive, as it should only modify globals and the file system the first time it's run now.

I think I disagree here. If `load` can only be called once, and it just returns the existing settings object otherwise, then why call `load` again instead of just using `settings` directly:

```diff
-const cfg = loadSettings({ defaults: {}, locked: {} });
+const cfg = settings;
```

If you prefer to use a function, at least define a separate function that takes no parameters.

As mentioned before, we should have an issue to refactor this for 1.10, not include it in the PR for 1.9.1.

**Review by jandubois — DISMISSED** [review]
I'm running the CI build on macOS, and the behaviour is unchanged: the lock icon is gone after closing and re-opening the main window, and then I can modify locked settings.

**Comment by ericpromislow** [comment]
I don't think we want to expose the `settings` object from `settings.ts`.

I could add a `getSettings` method that maybe requires `settings` to be initialized. Otherwise if it initializes it if it isn't set, it ends up doing the same thing as `settings.load()`

In the meantime, I think we can live with this change.

**Comment by jandubois** [comment]
> I don't think we want to expose the `settings` object from `settings.ts`.

I don't know, we expose `defaultSettings`, so why not `settings`?

> I could add a `getSettings` method that maybe requires `settings` to be initialized. Otherwise if it initializes it if it isn't set, it ends up doing the same thing as `settings.load()`

It seems silly to me to provide an accessor function for a global variable, but at least it can throw an error when the variable is not yet set.
 
> In the meantime, I think we can live with this change.

Yes, this is just about creating a follow-up issue to clean up the code.

**Review by jandubois — APPROVED** [review]
Change looks good to me, and worked correctly on macOS.

I would still like to see a second approval from another reviewer, and a follow-up issue filed to cleanup the code later.

File: pkg/rancher-desktop/config/settings.ts [hunk 1]
```diff
@@ -224,13 +224,24 @@ export async function clear() {
...
+  settings = undefined;
+}
+
 /**
  * Load the settings file or create it if not present.  If the settings have
  * already been loaded, return it without re-loading from disk.
  */
 export function load(deploymentProfiles: DeploymentProfileType): Settings {
+  if (settings) {
+    return settings;
```
Comment (mook-as) [inline pkg/rancher-desktop/config/settings.ts:241]: A comment here that this explicitly doesn't touch the deployment profiles would be nice.

**Review by mook-as — APPROVED** [review]

File: pkg/rancher-desktop/config/settings.ts (hunk 1 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/config/settings.ts:241]: Issue #5022 calls for refactoring, so such a comment won't be needed for long

**Comment by ericpromislow** [comment]
Followup issue written up at #5022

**Pull request merged**